src/main/java/org/cryptomator/cloudaccess/MetadataCachingProviderDecorator.java

```diff
@@ -191,9 +157,11 @@ private void evict(CloudPath cleartextPath) {
 		for (var path : cachedItemListRequests.asMap().keySet()) {
 			if (path.getKey().equals(cleartextPath)) {
 				cachedItemListRequests.invalidate(path);
-			} else if (path.getKey().getParent().equals(cleartextPath.getParent())) {
-				cachedItemListRequests.invalidate(path);
 			}
 		}
 	}
+
+	private void evict(Map.Entry<CloudPath, Optional<String>> folderKey) {
+		cachedItemListRequests.invalidate(folderKey);
+	}
```

Comment:
bad method name (as it is unclear whether this evicts items from `cachedItemListRequests` or `cachedItemListRequests`